<?xml version="1.0"?>
<launch>
    <arg name="x_pos" default="-2.0"/>
    <arg name="y_pos" default="0.0"/>
    <arg name="z_pos" default="0.0"/> 
    
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(find speed_limit_detect)/worlds/line_with_signs.world" />
        <arg name="paused" value="false"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="gui" value="true"/>
        <arg name="headless" value="false"/>
        <arg name="debug" value="false"/>
    </include>  

    <!--spawn turtlebot-->
    <param name="robot_description" command="$(find xacro)/xacro --inorder $(find speed_limit_detect)/urdf/turtlebot3_burger_pi.urdf.xacro" />
    <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf" args="-urdf -model turtlebot3_burger_pi -x $(arg x_pos) -y $(arg y_pos) -z $(arg z_pos) -param robot_description" />
  
    <!--robot state publisher-->
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" />

    <!--number classifier-->
    <rosparam param="template_images_dir">/home/tseng/tb3_ws/src/speed_limit_detect/images</rosparam>
    <node pkg="speed_limit_detect" type="speed_detect.py" name="speed_detect" />

    <!--motion control-->
    <rosparam param="camera_half_width">320</rosparam>
    <rosparam param="visualize_line_detection">true</rosparam>
    <rosparam param="use_speed_limit">true</rosparam>
    <node pkg="speed_limit_detect" type="line_follower.py" name="line_follower" />

    <!--rviz-->
    <arg name="rviz_config_file" value="$(find speed_limit_detect)/config/speed_detect.rviz"/>
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(arg rviz_config_file)"/>

</launch>